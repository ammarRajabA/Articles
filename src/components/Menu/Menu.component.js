import React, { Component } from 'react';

import './Menu.style.css'

export default class Menu extends Component{
	constructor(props){
		super(props);
		this.setWrapperRef = this.setWrapperRef.bind(this);
		this.handleClickOutside = this.handleClickOutside.bind(this);
	}
	componentDidMount(){
		document.addEventListener('mousedown', this.handleClickOutside);
	}
	componentWillUnmount(){
		document.removeEventListener('mousedown', this.handleClickOutside);
	}
	setWrapperRef(node) {
		this.wrapperRef = node;
	}
	handleClickOutside(event) {
		if (this.wrapperRef && !this.wrapperRef.contains(event.target)) {
			this.props.onDismiss()
		}
	}
	render(){
		if (!this.props.show) return <span></span>
		return(
			<div className="menu-popup-container" ref={this.setWrapperRef}>
				{
					Object.keys(this.props.links).map((link,index)=>
							<span key={index.toString()} className="menu-option" onClick={this.props.links[link]}>{link}</span>
						)
				}
			</div>
			)
	}
}