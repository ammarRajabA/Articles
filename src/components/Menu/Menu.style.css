.menu-popup-container{
	position: absolute;
	background-color: #FFFFFF;
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	justify-content: center;
	box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2), 0 3px 10px 0 rgba(0,0,0,0.19);
	z-index: 1;
	padding-right:20px;
	right: 10px;
	width: 150px;
}
.menu-popup-container .menu-option{
	color: #333333;
	width: 100%;
	cursor: pointer;
	transition: 0.3s;
	padding-left:20px;
	padding-right:20px;
	padding-top: 10px;
	padding-bottom: 10px;
	font-size: 12px;
	border-bottom: 1px solid #E3E3E3;
}
.menu-popup-container .menu-option:hover{
	background-color: #E1E1E1
}