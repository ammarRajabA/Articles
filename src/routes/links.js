import history from './history';

export default {
	"Home":()=>history.push('/'),
	"Articles":()=>history.push('/articles'),
	"New Article":()=>history.push('/new-article'),
	"About":()=>history.push('/about')
}