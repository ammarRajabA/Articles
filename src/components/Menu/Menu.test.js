import React from 'react'
import { shallow ,mount } from 'enzyme';

import Menu from './Menu.component';
import links from '../../routes/links'

describe('Menu component',function () {
	it('match snapshot',function () {
		const component = mount(<Menu links={links} show={true}/>);
		expect(component).toMatchSnapshot();
	})
})