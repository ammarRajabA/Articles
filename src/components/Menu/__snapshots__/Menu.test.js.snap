// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component match snapshot 1`] = `
<Menu
  links={
    Object {
      "About": [Function],
      "Articles": [Function],
      "Home": [Function],
      "New Article": [Function],
    }
  }
  show={true}
>
  <div
    className="menu-popup-container"
  >
    <span
      className="menu-option"
      key="0"
      onClick={[Function]}
    >
      Home
    </span>
    <span
      className="menu-option"
      key="1"
      onClick={[Function]}
    >
      Articles
    </span>
    <span
      className="menu-option"
      key="2"
      onClick={[Function]}
    >
      New Article
    </span>
    <span
      className="menu-option"
      key="3"
      onClick={[Function]}
    >
      About
    </span>
  </div>
</Menu>
`;
